pkg/controllers/machine_controller.go

```diff
@@ -790,6 +797,34 @@ func (r *MachineReconciler) setTCMallocPath(domain *libvirtxml.Domain) error {
 	return nil
 }
 
+func (r *MachineReconciler) setGuestAgent(machine *api.Machine, domainDesc *libvirtxml.Domain) {
+	if domainDesc.Devices == nil {
+		domainDesc.Devices = &libvirtxml.DomainDeviceList{}
+	}
+
+	if domainDesc.Devices.Channels == nil {
+		domainDesc.Devices.Channels = make([]libvirtxml.DomainChannel, 0, 1)
+	}
+
+	socketPath := filepath.Join(r.host.MachineDir(machine.GetID()), "qemu-guest-agent.sock")
+	agent := libvirtxml.DomainChannel{
+		Source: &libvirtxml.DomainChardevSource{
+			UNIX: &libvirtxml.DomainChardevSourceUNIX{
+				Mode: "bind",
+				Path: socketPath,
+			},
+		},
+		Target: &libvirtxml.DomainChannelTarget{
+			VirtIO: &libvirtxml.DomainChannelTargetVirtIO{
+				Name: "org.qemu.guest_agent.0",
+			},
+		},
+	}
+
+	domainDesc.Devices.Channels = append(domainDesc.Devices.Channels, agent)
+	machine.Status.GuestAgentStatus = &api.GuestAgentStatus{Type: machine.Spec.GuestAgent, Addr: "unix://" + socketPath}
```

Comment:
We do not use this property 
```suggestion
	machine.Status.GuestAgentStatus = &api.GuestAgentStatus{ Addr: "unix://" + socketPath}
```

Reply:
it is information for user. Do you really want to delete it?

Reply:
wait, i used status.Type in getShutdownMode

Reply:
fixed